### PR TITLE
Add first attempt to get the golden image for FreeBSD.

### PR DESCRIPTION
**Pull request by krionbsd**

### CODE REVIEW

**Comment by bryceml** [comment]
We'll see how far this gets, and we can make changes incrementally.

File: os-images/AWS/freebsd/freebsd-12-us-west-2.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (bryceml) [inline os-images/AWS/freebsd/freebsd-12-us-west-2.json:1]: I think we will have to rename this file to have 12.1 in the name of the file.

**Comment by krionbsd** [comment]
re-run full all

**Comment by krionbsd** [comment]
> Not sure why the tests are failing here...

Yeah, I'm on it right now

File: state-tree/golden-images-provision.sls [hunk 2]
```diff
@@ -49,6 +49,11 @@ include:
   - sed
   {%- endif %}
   {%- if grains['os'] not in ('MacOS', 'Windows') %}
+  {%- if grains['os_family'] in ('FreeBSD') %}
```
Comment (s0undt3ch) [inline state-tree/golden-images-provision.sls:52]: `in ('FreeBSD',)`

File: .ci/freebsd-12.1 [hunk 3]
```diff
@@ -0,0 +1,35 @@
+@Library('salt@master-1.8-golden') _
+
+// Define the maximum time, in hours, that a test run should run for
+def testrun_timeout = 6
+
+def distro_name = 'freebsd'
+def distro_version = '12.1'
```
Comment (bryceml) [inline .ci/freebsd-12.1:7]: here

File: os-images/AWS/freebsd/freebsd-12.1-us-west-2.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "_comment": "Build with `packer build -var-file=freebsd-12-us-west-2.json freebsd.json`",
+  "ssh_username": "ec2-user",
+  "aws_region": "us-west-2",
+  "os_name": "FreeBSD",
+  "os_version": "12.1",
```
Comment (bryceml) [inline os-images/AWS/freebsd/freebsd-12.1-us-west-2.json:6]: here

File: os-images/AWS/freebsd/freebsd-12.1-us-west-2.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  "salt_provision_version": "3000.3",
+  "salt_provision_root_dir": "/tmp/salt-provision",
+  "salt_provision_python_version": "3.6.10",
+  "salt_provision_pyenv_deps": "",
+  "state_name": "golden-images-provision",
+  "ami_filter": "FreeBSD 12.1-RELEASE-amd64-*",
+  "ami_owner": "679593333241",
+  "ami_name_prefix": "saltstack",
+  "ami_name_suffix": "freebsd/12.1",
```
Comment (bryceml) [inline os-images/AWS/freebsd/freebsd-12.1-us-west-2.json:15]: here

File: os-images/AWS/freebsd/freebsd-12.1-us-west-2.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (bryceml) [inline os-images/AWS/freebsd/freebsd-12.1-us-west-2.json:1]: the filename of this file.

File: .ci/freebsd-12.1 [hunk 7]
```diff
@@ -0,0 +1,35 @@
+@Library('salt@master-1.8-golden') _
```
Comment (bryceml) [inline .ci/freebsd-12.1:1]: the filename of this file

File: os-images/AWS/freebsd/freebsd.json [hunk 8]
```diff
@@ -0,0 +1,150 @@
+{
+  "_comment": "Build with `packer build ubuntu.json`",
```
Comment (bryceml) [inline os-images/AWS/freebsd/freebsd.json:2]: I would change this to be say freebsd.json.

Comment (bryceml) [reply os-images/AWS/freebsd/freebsd.json:2]: This still needs changed.

**Comment by bryceml** [comment]
> We should just merge this.
> Anyone against?

We can see the kitchen run in the pr, I think it would be easiest to debug here.

**Comment by krionbsd** [comment]
re-run all

**Comment by s0undt3ch** [comment]
@bryceml are you good with the PR now?

**Comment by bryceml** [comment]
@s0undt3ch yeah, looks like there are only 20 failures.  My main concern was getting past the point where we install the nox requirements.

**Comment by s0undt3ch** [comment]
@bryceml ok. I'll leave the merge up to you.

**Pull request merged**